metal/resource_metal_reserved_ip_block.go

```diff
@@ -49,6 +55,19 @@ func metalIPComputedFields() map[string]*schema.Schema {
 			Type:     schema.TypeBool,
 			Computed: true,
 		},
+		"wait_for_state": {
```

Comment:
Makes more sense to define `wait_for_state` directly in resourceMetalReservedIPBlock() since it's not computed. 

(This should be managed in a different issue/pr) Even more, I'm not sure why there is a distinction for some computed params defined in two different function metalIPResourceComputedFields() and metalIPComputedFields(). Having a single function would simplify code maintaining  and avoid errors, for example there I found `address_family` is defined twice. 
https://github.com/equinix/terraform-provider-metal/blob/3834e17fb84992b49c8b758faf358bd6ff852867/metal/resource_metal_reserved_ip_block.go#L26

https://github.com/equinix/terraform-provider-metal/blob/3834e17fb84992b49c8b758faf358bd6ff852867/metal/resource_metal_reserved_ip_block.go#L76

Reply:
I think the distinction owes to packngo's definition of `IPAddressReservation` and `IPAddressCommon`. And this owes to the API spec and (more importantly API responses) where reservations (user requested IP blocks) have different properties than non-reservations (dynamically or statically assigned addresses that the user can reclaim/reuse).

Reply:
That said, if there are overlapping aspects of the code that can be refactored, we should look into that.

Reply:
I've removed the duplicate definition.

This helper shares schema definitions between metal_ip_attachment, data.metal_precreated_ip_block, metal_resource_ip_block. This is something we can revisit later.